src/linker/Mono.Linker.csproj

```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(MonoBuild)' == ''">
+    <TargetFramework>net5.0</TargetFramework>
```

Comment:
https://github.com/mono/linker/blob/master/src/linker/Directory.Build.props#L13 sets the `TargetFramework` for this and the ref project - either update it there, or remove it if you prefer to set it in each project.